ckanserviceprovider/db.py

```diff
@@ -514,18 +514,61 @@ def _get_metadata(job_id):
     return metadata
 
 
-def _get_logs(job_id):
+def _get_logs(job_id, limit=None):
     """Return any logs for the given job_id from the logs table."""
     # Avoid SQLAlchemy "Unicode type received non-unicode bind param value"
     # warnings.
     job_id = unicode(job_id)
-
-    results = ENGINE.execute(
-        LOGS_TABLE.select().where(LOGS_TABLE.c.job_id == job_id)).fetchall()
+    try:
+        int(limit)
+        limit_is_valid = True
+    except (ValueError, TypeError):
+        # None or "one" (or similar)
+        limit_is_valid = False
+
+    if not limit_is_valid:
+        results = ENGINE.execute(
+            LOGS_TABLE.select().where(LOGS_TABLE.c.job_id == job_id)).fetchall()
+    else:
+        results = ENGINE.execute(
+            LOGS_TABLE.select().where(LOGS_TABLE.c.job_id == job_id).\
+                order_by(LOGS_TABLE.c.timestamp.desc()).limit(limit)).fetchall()
 
     results = [dict(result) for result in results]
 
     for result in results:
         result.pop("job_id")
 
     return results
+
+
+def add_logs(job_id, message=None, level=None,
```

Comment:
I had to add custom function for tests as there is no way to insert logs from ckan-service-provider